### PR TITLE
Fixing pyephem dependent tests

### LINKED ISSUE #80: Failing tests due to missing pyephem
Some tests are failing due to the missing pyephem package. They shouldn't really as it's listed only as an optional dependency:

```
=========================================================== test session starts ============================================================
platform darwin -- Python 2.7.10 -- pytest-2.5.1

Running tests with Astropy version 1.1.dev13304.
Running tests in astroplan /Users/bsipocz/munka/devel/astroplan/docs.

Platform: Darwin-14.5.0-x86_64-i386-64bit

Executable: /Users/bsipocz/.virtualenvs/astropy-dev/bin/python

Full Python Version: 
2.7.10 (default, May 26 2015, 20:29:34) 
[GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)]

encodings: sys: ascii, locale: UTF-8, filesystem: utf-8, unicode bits: 15
byteorder: little
float info: dig: 15, mant_dig: 15

Numpy: 1.9.2
Scipy: 0.16.0
Matplotlib: 1.4.3
pyephem: not available

plugins: mpl
collected 62 items 

astroplan/core.py .........F.........
astroplan/sites.py .
astroplan/plots/tests/test_sky.py F
astroplan/tests/test_core.py .s......................F.
astroplan/tests/test_moon.py F
astroplan/tests/test_sites.py .....
astroplan/tests/test_utils.py s
../../../../../../../../Users/bsipocz/munka/devel/astroplan/docs/api.rst .
../../../../../../../../Users/bsipocz/munka/devel/astroplan/docs/getting_started.rst .
../../../../../../../../Users/bsipocz/munka/devel/astroplan/docs/index.rst .
../../../../../../../../Users/bsipocz/munka/devel/astroplan/docs/installation.rst .
../../../../../../../../Users/bsipocz/munka/devel/astroplan/docs/tutorials/iers.rst .
../../../../../../../../Users/bsipocz/munka/devel/astroplan/docs/tutorials/index.rst .
../../../../../../../../Users/bsipocz/munka/devel/astroplan/docs/tutorials/plots.rst .
../../../../../../../../Users/bsipocz/munka/devel/astroplan/docs/tutorials/summer_triangle.rst .

================================================================= FAILURES =================================================================
_______________________________________________ [doctest] astroplan.core.Observer.moon_altaz _______________________________________________
1497         Examples
1498         --------
1499         Calculate the altitude and azimuth of the moon at Apache Point
1500         Observatory:
1501 
1502         >>> from astroplan import Observer
1503         >>> from astropy.time import Time
1504         >>> apo = Observer.at_site("APO")
1505         >>> time = Time("2015-08-29 18:35")
1506         >>> altaz_moon = apo.moon_altaz(time)
UNEXPECTED EXCEPTION: ImportError(u'The moon_altaz function currently requires PyEphem to compute the position of the moon.',)
Traceback (most recent call last):

  File "/sw/lib/python2.7/doctest.py", line 1315, in __run
    compileflags, 1) in test.globs

  File "<doctest astroplan.core.Observer.moon_altaz[4]>", line 1, in <module>

  File "astroplan/core.py", line 1516, in moon_altaz
    raise ImportError("The moon_altaz function currently requires "

ImportError: The moon_altaz function currently requires PyEphem to compute the position of the moon.

astroplan/core.py:1506: UnexpectedException
_____________________________________________________________ test_moon_altaz ______________________________________________________________

    def test_moon_altaz():
        time = Time('2012-06-21 03:00:00')
        location = EarthLocation.from_geodetic(-155*u.deg, 19*u.deg, 0*u.m)
        obs = Observer(location=location, pressure=0*u.bar)
>       altaz = obs.moon_altaz(time)

astroplan/tests/test_core.py:1102: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <Observer: location (lon, lat, el)=(-155.0 deg, 19.0 deg, 0.0 m),
    timezone=<UTC>,
    pressure=<Quantity 0.0 bar>>
time = <Time object: scale='utc' format='iso' value=2012-06-21 03:00:00.000>

    def moon_altaz(self, time):
        """
            Returns the position of the moon in alt/az.
    
            TODO: Currently `moon_altaz` uses PyEphem to calculate the position
            of the moon.
    
            Parameters
            ----------
            time : `~astropy.time.Time` or other (see below)
                This will be passed in as the first argument to
                the `~astropy.time.Time` initializer, so it can be anything that
                `~astropy.time.Time` will accept (including a `~astropy.time.Time`
                object).
    
            Returns
            -------
            altaz : `~astropy.coordinates.SkyCoord`
                Position of the moon transformed to altitude and azimuth
    
            Examples
            --------
            Calculate the altitude and azimuth of the moon at Apache Point
            Observatory:
    
            >>> from astroplan import Observer
            >>> from astropy.time import Time
            >>> apo = Observer.at_site("APO")
            >>> time = Time("2015-08-29 18:35")
            >>> altaz_moon = apo.moon_altaz(time)
            >>> print("alt: {0.alt}, az: {0.az}".format(altaz_moon)) # doctest: +FLOAT_CMP
            alt: -64.1659594407 deg, az: 345.360401117 deg
            """
        if not isinstance(time, Time):
            time = Time(time)
    
        try:
            import ephem
        except ImportError:
>           raise ImportError("The moon_altaz function currently requires "
                              "PyEphem to compute the position of the moon.")
E           ImportError: The moon_altaz function currently requires PyEphem to compute the position of the moon.

astroplan/core.py:1516: ImportError
____________________________________________________________ test_illumination _____________________________________________________________

    def test_illumination():
        time = Time(['1990-01-01 00:00:00', '1990-03-01 06:00:00',
                     '1990-06-01 12:00:00', '1990-11-01 18:00:00'])
        location = EarthLocation.from_geodetic(-155*u.deg, 19*u.deg, 0*u.m)
        obs = Observer(location)
        # Get illumination via time
>       illumination1 = obs.moon_illumination(time)

astroplan/tests/test_moon.py:17: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <Observer: location (lon, lat, el)=(-155.0 deg, 19.0 deg, 0.0 m),
    timezone=<UTC>>
time = <Time object: scale='utc' format='iso' value=['1990-01-01 00:00:00.000' '1990-03-01 06:00:00.000'
 '1990-06-01 12:00:00.000' '1990-11-01 18:00:00.000']>

    def moon_illumination(self, time):
        """
            Calculate the illuminated fraction of the moon.
    
            Parameters
            ----------
            time : `~astropy.time.Time` or other (see below)
                This will be passed in as the first argument to
                the `~astropy.time.Time` initializer, so it can be anything that
                `~astropy.time.Time` will accept (including a `~astropy.time.Time`
                object).
    
            moon : `~astropy.coordinates.SkyCoord` or `None` (default)
                Position of the moon at time ``time``. If `None`, will calculate
                the position of the moon with `~astroplan.moon.get_moon`.
    
            sun : `~astropy.coordinates.SkyCoord` or `None` (default)
                Position of the sun at time ``time``. If `None`, will calculate
                the position of the Sun with `~astropy.coordinates.get_sun`.
    
            Returns
            -------
            float
                Fraction of lunar surface illuminated
    
            Examples
            --------
            How much of the lunar surface is illuminated at 2015-08-29 18:35 UTC,
            which we happen to know is the time of a full moon?
    
            >>> from astroplan import Observer
            >>> from astropy.time import Time
            >>> apo = Observer.at_site("APO")
            >>> time = Time("2015-08-29 18:35")
            >>> apo.moon_illumination(time) # doctest: +SKIP
            array([ 0.99972487])
            """
        if not isinstance(time, Time):
            time = Time(time)
    
>       return moon_illumination(time, self.location)

astroplan/core.py:1430: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

time = <Time object: scale='utc' format='iso' value=['1990-01-01 00:00:00.000' '1990-03-01 06:00:00.000'
 '1990-06-01 12:00:00.000' '1990-11-01 18:00:00.000']>
location = <EarthLocation (-5467562.495834137, -2549566.2632485162, 2063349.463036347) m>

    def moon_illumination(time, location):
        """
        Calculate fraction of the moon illuminated
    
        Parameters
        ----------
        time : `~astropy.time.Time`
            Time of observation
    
        location : `~astropy.coordinates.EarthLocation`
            Location of observer
    
        Returns
        -------
        k : float
            Fraction of moon illuminated
        """
>       i = moon_phase_angle(time, location)

astroplan/moon.py:144: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

time = <Time object: scale='utc' format='iso' value=['1990-01-01 00:00:00.000' '1990-03-01 06:00:00.000'
 '1990-06-01 12:00:00.000' '1990-11-01 18:00:00.000']>
location = <EarthLocation (-5467562.495834137, -2549566.2632485162, 2063349.463036347) m>

    def moon_phase_angle(time, location):
        """
        Calculate lunar orbital phase [radians].
    
        Parameters
        ----------
        time : `~astropy.time.Time`
            Time of observation
    
        location : `~astropy.coordinates.EarthLocation`
            Location of observer
    
        Returns
        -------
        i : float
            Phase angle of the moon [radians]
        """
        # TODO: cache these sun/moon SkyCoord objects
        sun = get_sun(time).transform_to(AltAz(location=location, obstime=time))
>       moon = get_moon(time, location)

astroplan/moon.py:116: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

time = <Time object: scale='utc' format='iso' value=['1990-01-01 00:00:00.000' '1990-03-01 06:00:00.000'
 '1990-06-01 12:00:00.000' '1990-11-01 18:00:00.000']>
location = <EarthLocation (-5467562.495834137, -2549566.2632485162, 2063349.463036347) m>, pressure = None

    def get_moon(time, location, pressure=None):
        """
        Position of the Earth's moon.
    
        Currently uses PyEphem to calculate the position of the moon by default
        (requires PyEphem to be installed). Set ``use_pyephem`` to `False` to
        calculate the moon position with jplephem (requires jplephem to be
        installed).
    
        Parameters
        ----------
        time : `~astropy.time.Time` or see below
            Time of observation. This will be passed in as the first argument to
            the `~astropy.time.Time` initializer, so it can be anything that
            `~astropy.time.Time` will accept (including a `~astropy.time.Time`
            object).
    
        location : `~astropy.coordinates.EarthLocation`
            Location of the observer on Earth
    
        pressure : `None` or `~astropy.units.Quantity` (optional)
    
        Returns
        -------
        moon_sc : `~astropy.coordinates.SkyCoord`
            Position of the moon at ``time``
        """
        if not isinstance(time, Time):
            time = Time(time)
    
        try:
            import ephem
        except ImportError:
>           raise ImportError("The get_moon function currently requires "
                              "PyEphem to compute the position of the moon.")
E           ImportError: The get_moon function currently requires PyEphem to compute the position of the moon.

astroplan/moon.py:66: ImportError
```

### PR DESCRIPTION
**Pull request by bsipocz**
Fixes #80

### CODE REVIEW

File: astroplan/core.py [hunk 1]
```diff
@@ -26,10 +26,11 @@
            "Constraint", "TimeWindow", "AltitudeRange",
            "AboveAirmass", "MAGIC_TIME"]
 
-#__doctest_requires__ = {'*': ['scipy.integrate']}
```
Comment (cdeil) [inline astroplan/core.py]: Can you please leave a `TODO` comment here to make it easier to update this in the future?
```
# TODO: remove this statement once the moon is implemented without pyephem
```

File: astroplan/tests/test_moon.py [hunk 2]
```diff
@@ -26,6 +35,8 @@ def test_illumination():
 
     assert_allclose(illumination1, pyephem_illumination, atol=0.05)
 
+
+@pytest.mark.skipif('not HAS_PYEPHEM')
```
Comment (cdeil) [inline astroplan/tests/test_moon.py]: Remove this pytest mark.
pytest marks are only useful for test functions, and this is not a test function.

**Comment by cdeil** [comment]
Two minor inline comments, otherwise :+1: .

(Well, I'm not sure it's useful to take the `pyephem` import out of the `print_` functions and move it to the top of the file, but it doesn't really matter, either way is fine with me.)

**Comment by bsipocz** [comment]
Sure, the ``print_`` decorator remained in there by mistake (due to the confusion I mentioned in #80).

**Comment by cdeil** [comment]
Thanks!

**Pull request merged**